src/Infrastructure/Service/LifecycleManager.php

```diff
@@ -81,27 +89,25 @@ public function update(): void
     }
 
     /**
-     * @throws \SprykerSdk\Sdk\Infrastructure\Exception\SdkVersionNotFoundException
-     *
      * @return array<\SprykerSdk\SdkContracts\Entity\MessageInterface>
      */
     public function checkForUpdate(): array
     {
+        if ($this->environment !== 'prod') {
+            return [new Message('Not possible to check updates for development version.')];
+        }
+
         $versionFilePath = $this->sdkDirectory . '/VERSION';
 
         if (!is_file($versionFilePath)) {
-            throw new SdkVersionNotFoundException(
-                sprintf('Could not find %s file, skip updatable check', $versionFilePath),
-            );
+            return [new Message('Could not find `VERSION` file, skip updatable check')];
```

Comment:
Minor: move `VERSION` into the constant and use it in messages and in `$versionFilePath` var